Mathlib/Tactic/ByContra.lean

```diff
@@ -40,8 +40,8 @@ example : 1 < 2 := by
 syntax (name := byContra!) "by_contra!" (ppSpace colGt binderIdent)? Term.optType : tactic
 
 macro_rules
-  | `(tactic| by_contra!%$tk $[_%$under]? $[: $ty]?) =>
-    `(tactic| by_contra! $(mkIdentFrom (under.getD tk) `this (canonical := true)):ident $[: $ty]?)
+  | `(tactic| by_contra! $[_%$under]? $[: $ty]?) =>
+    `(tactic| by_contra! $(mkIdent `this):ident $[: $ty]?)
```

Comment:
This loses the hover information for `by_contra! _` on the underscore.

How about for clarity we split this up into two rules
```
  | `(tactic| by_contra! $[: $ty]?) =>
    `(tactic| by_contra! $(mkIdent `this):ident $[: $ty]?)
  | `(tactic| by_contra! _%$under $[: $ty]?) =>
    `(tactic| by_contra! $(mkIdentFrom under `this):ident $[: $ty]?)
```
Alternatively, something like this would work:
```
  | `(tactic| by_contra! $[_%$under]? $[: $ty]?) =>
    `(tactic| by_contra! $(mkIdentFrom (under.getD .missing) `this):ident $[: $ty]?)
```

Reply:
Kyle, thanks for the suggestions: I went for the more explicit case split.